rfc/extend-misconfiguration-scanners.md

```diff
@@ -0,0 +1,106 @@
+However, some scanners cannot directly categorize findings using a single or existing scope which can limit integration options.
+This behavior, along with the lack of dynamic- and multi-scope options, also underlines an important limitation of how findings are being described, categorized, processed, and analyzed.
+Note that this RFC does not intend to resolve this behavior, but rather draw attention to it.
+
+The integration of [CIS Docker Benchmark](https://github.com/goodwithtech/dockle) scanner requires additional changes to address the scope-based categorization limitations.
+In KubeClarity, the scanner defines its own [API model](https://github.com/openclarity/kubeclarity/blob/5ac3048b7a782c900a9bef846a91a7735ba77e24/api/swagger.yaml#L243C26-L243C26) to describe related security findings.
+This makes the migration of scanning logic to VMClarity problematic for two main reasons:
+
+- Logic in the form of a new independent scanner family does not conform to any existing *security scopes*.
+CIS Docker Benchmark provides little benefit on its own due to scope constraints currently defined for the scanning logic.
+
+- Logic is *too specific* and *provider-dependant* to be part of an existing scanner family.
+CIS Docker Benchmark scan results cannot be uniformly converted to other findings without some loss of data.
+
+## Proposal
+
+The CIS Docker Benchmark scanner can be migrated as part of **misconfiguration scanner family** to enrich the findings on assets with additional security coverage.
+Contextually, the misconfiguration findings serve as a superset of CIS Docker Benchmark results.
+This approach benefits VMClarity in several ways:
+
+* The misconfiguration findings can be generalized and reused without impacting the existing scopes
+
+The misconfiguration [API model](https://github.com/openclarity/vmclarity/blob/bfc32ec88ee266157aaf7bcae7b17c4b2ee5c868/api/openapi.yaml#L3083) is not abstract enough to enable integration of new scanners.
+Minor API changes are required to make the model more generic and enable direct conversion of CIS Docker Benchmark results.
+This also standardizes the model for usage and simplifies future integrations.
+
+- The misconfiguration scanner family enables an idiomatic way to migrate the required scanning logic from KubeClarity
+
+Integrating the CIS Docker Benchmark can be accomplished by reusing the existing patterns to minimize changes.
+The migration can then be performed as an implementation of a scanner within the existing misconfiguration family.
+
+### Abandoned Ideas (Optional)
+
+---
+
+## Implementation
+
+1. Extend the misconfiguration model to be more generic
+
+The model preserves most of its original properties but is generalized to enable direct conversion from CIS Docker Benchmark results.
+
+```yaml
+Misconfiguration:
+  type: object
+  properties:
+    scannerName: # preserved
+      type: string
+    code: # replaces `testID`; maps CISDockerBenchmarkResultsEX.code
```

Comment:
`code` is not very intuitive for me here but tbh I do not have a better suggestion 😞 `testID` or something like `checkID` are also not the best options here.

Your description is clear anyway 😄

Reply:
I agree, left it as `code` due to a lack of a better alternative. `checkID` makes more sense to me, but I wanted to avoid the `ID` similar to other findings.

Reply:
"TestID" was picked because it was somewhat generic, i.e. Lynis might have an ID like "DB034" and Docker Bench has something like "1.1.4". In all cases its a specific to the scanner i.e. Scanner + TestID uniquely identifies a specific test (does a file exist, or is a config set on github) that failed.

This was something we wanted to revisit because if two scanners in the system detect the same issue, they are reported as different findings (essentially every test for every unique location for every scanner that fails is a unique finding). Ideally we should be able to merge them together somehow, for example if two scanners find that a file as two many permissions then it could be a Finding structured like:

All failures for a specific file are bundled together:

```
Misconfiguration:
   Location: <file path>
   FailedTests:
      - Scanner: XYZ
         TestID: FooBar
      - Scanner: Ham
         TestID: Cheese
      - Scanner: Ham
         TestID: Mayo
```
       
Or alternatively we somehow understand which tests are for a specific misconfiguration and then bundle them per name and location:

```
Misconfiguration:
   MisconfigurationName: Too Many Permissions
   Location: <file path>
   Foundby:
      - Scanner: XYZ
         TestID: FooBar
      - Scanner: Ham
         TestID: Cheese
      - Scanner: Ham
         TestID: Mayo

Reply:
A similar approach can be reused for other findings as well @Tehsmash. I also have concerns regarding findings in general due to these limitations (merging issues, data duplication, provider dependencies, extension limitations, etc) which I am currently working on. Thanks a lot for addressing this @Tehsmash!

The plan here is to integrate the CIS Docker quickly with minimal changes to the API, while the other proposal will tackle these finding-related challenges in one go.

Reply:
A similar approach can be reused for other findings as well. I also have concerns regarding findings in general due to these limitations (merging, data duplication, provider dependencies, extension limitations, etc) which I am currently working on. Thanks a lot for addressing this @Tehsmash!

The plan for this proposal is to integrate the CIS Docker quickly with minimal changes to the API, while the other proposal will tackle these finding-related challenges in one go.

Reply:
I'd use plain `id` instead of `code`. Latter is kinda misleading.